core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -202,7 +202,7 @@ private[spark] class PythonRDD(
       this.interrupt()
     }
 
-    override def run(): Unit = Utils.logUncaughtExceptions {
+    override def run(): Unit = Utils.tryLog {
```

Comment:
btw this happened a few times in the past in Spark. somebody added a return in a giant block that is wrapped in a closure. Code reviewers couldn't catch it because it didn't show up in the diff.